### ISSUE TITLE
Make it more apparent if a part needs to be configured

### ISSUE THREAD

**Issue by Piedone** [OrchardCMS/OrchardCore#6105]
Currently when you add a part with settings then the only indicator of that part being configurable is the "Edit" button in the content type editor. For parts where they won't work at all without configuration, like `WidgetsListPart`, I think this is too evasive. If the user doesn't know already how that part works they can easily miss (the apparently optional) step of editing settings, getting confused why something doesn't work.

I don't know how this can be improved nicely. Perhaps a notification asking you to edit the part settings if it has one?

**Comment by jptissot** [comment]
I agree, often, our developers setup a fields and whey they try to edit the content, they realize the editor is not set to WYSIWYG. Maybe we need to display some settings in the list of fields / parts of a ContentType (in the white space). Maybe they could event be clickable to edit the settings ?

**Comment by Piedone** [comment]
We could have a collapsible settings area for every field/part in the content type editor, much like in O1. However, this would be a major shift from the current implementation.

**Comment by sebastienros** [comment]
WE have talked about rewriting the whole content type editor to make it more dynamic, single page. So you can click on buttons to expand sections without losing the current context (same page).
@agriffard will link to the issue ;)

**Comment by agriffard** [comment]
Related to #472

**Comment by agriffard** [comment]
Suggestion: Add an overridable `NeedsConfiguration` property, default to false.

Ex for ListPart: Return `true` if no `Content Types` are selected.
Ex for MultiTexField: Return `true` if list of values is not defined.

If NeedsConfiguration => Display a visual hint in Content Type Edit page.

**Comment by gvkries** [comment]
@Piedone We now have "Add & Configure" buttons in the list. Does it solve this?

**Comment by Piedone** [comment]
Yeah, this is good enough, thanks.
